packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/update.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  SavedObjectsErrorHelpers,
+  type SavedObject,
+  type SavedObjectSanitizedDoc,
+  SavedObjectsRawDoc,
+  SavedObjectsRawDocSource,
+} from '@kbn/core-saved-objects-server';
+import { SavedObjectsUtils } from '@kbn/core-saved-objects-utils-server';
+import { encodeHitVersion } from '@kbn/core-saved-objects-base-server-internal';
+import {
+  SavedObjectsUpdateOptions,
+  SavedObjectsUpdateResponse,
+} from '@kbn/core-saved-objects-api-server';
+import { DEFAULT_REFRESH_SETTING, DEFAULT_RETRY_COUNT } from '../constants';
+import { getCurrentTime, getExpectedVersionProperties } from './utils';
+import { ApiExecutionContext } from './types';
+import { PreflightCheckNamespacesResult } from './helpers';
+
+export interface PerformUpdateParams<T = unknown> {
+  type: string;
+  id: string;
+  attributes: T;
+  options: SavedObjectsUpdateOptions<T>;
+}
+
+export const performUpdate = async <T>(
```

Comment:
Implementing partial updates and all the changes involved there is going to be a much more pleasant dev exp!